### PR TITLE
NMS-15342: Allow property extenders to index by ifIndex

### PR DESCRIPTION
**Pull request by amay94**
Interface-level resources are given the resourceType `if,` which is used as a reference in thresholding config files and possibly elsewhere. Rather than changing that, this just adjusts SnmpPropertyExtender matching to allow `ifIndex` to match with `if` resources.

### External References

* Jira (Issue Tracker): https://opennms.atlassian.net/browse/NMS-15342

### CODE REVIEW

**Review by aramosvizcarra — DISMISSED** [review]
Looks good to me, I would try to add a test for this (if possible)

File: features/collection/snmp-collector/src/main/java/org/opennms/netmgt/collectd/SnmpPropertyExtenderProcessor.java [hunk 1]
```diff
@@ -94,7 +95,13 @@ public void process(final SnmpCollectionSet collectionSet, final String collecti
             // Apply MIB Properties
             collectionSet.getResources().forEach(r -> {
                 mibObjProperties.forEach(p -> {
-                    if (p.getInstance().equals(r.getResourceTypeName())) {
+                    // Property extenders should be able to index by ifIndex similar to how other mibObjs do. See
+                    // MNS-15342.
```
Comment (christianpape) [inline features/collection/snmp-collector/src/main/java/org/opennms/netmgt/collectd/SnmpPropertyExtenderProcessor.java:99]:
```suggestion
                    // NMS-15342.
```

**Review by christianpape — APPROVED** [review]
LGTM.

**Review by christianpape — APPROVED** [review]
LGTM.

**Pull request merged**